electrum/wizard.py

```diff
@@ -652,16 +654,13 @@ class ServerConnectWizard(AbstractWizard):
     def __init__(self, daemon: 'Daemon'):
         AbstractWizard.__init__(self)
         self.navmap = {
-            'autoconnect': {
-                'next': 'server_config',
+            'welcome': {
+                'next': lambda d: 'proxy_config' if d['want_proxy'] else 'server_config',
                 'accept': self.do_configure_autoconnect,
-                'last': lambda d: d['autoconnect']
-            },
-            'proxy_ask': {
-                'next': lambda d: 'proxy_config' if d['want_proxy'] else 'autoconnect'
+                'last': lambda d: bool(d['autoconnect'] and not d['want_proxy'])
             },
             'proxy_config': {
-                'next': 'autoconnect',
+                'next': 'server_config',
```

Comment:
If someone checks `config_proxy` it does not matter whether they check `config_server`, they will see both screens.